src/main/java/de/fearnixx/jeak/teamspeak/voice/connection/VoiceConnectionService.java

```diff
@@ -0,0 +1,128 @@
+package de.fearnixx.jeak.teamspeak.voice.connection;
+
+import com.github.manevolent.ts3j.identity.LocalIdentity;
+import de.fearnixx.jeak.IBot;
+import de.fearnixx.jeak.event.bot.IBotStateEvent;
+import de.fearnixx.jeak.reflect.FrameworkService;
+import de.fearnixx.jeak.reflect.Inject;
+import de.fearnixx.jeak.reflect.Listener;
+import de.fearnixx.jeak.service.event.IEventService;
+import de.fearnixx.jeak.service.teamspeak.IUserService;
+import de.fearnixx.jeak.teamspeak.IServer;
+import de.fearnixx.jeak.teamspeak.voice.connection.info.AbstractVoiceConnectionInformation;
+import de.fearnixx.jeak.teamspeak.voice.connection.info.ConfigVoiceConnectionInformation;
+import de.fearnixx.jeak.teamspeak.voice.connection.info.DbVoiceConnectionInformation;
+import de.fearnixx.jeak.voice.connection.IVoiceConnection;
+import de.fearnixx.jeak.voice.connection.IVoiceConnectionPool;
+import de.fearnixx.jeak.voice.connection.IVoiceConnectionService;
+import de.mlessmann.confort.LoaderFactory;
+import de.mlessmann.confort.config.FileConfig;
+
+import java.io.File;
+import java.security.GeneralSecurityException;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Consumer;
+
+@FrameworkService(serviceInterface = IVoiceConnectionService.class)
+public class VoiceConnectionService implements IVoiceConnectionService {
+
+    @Inject
+    private IServer server;
+
+    @Inject
+    private IBot bot;
+
+    @Inject
+    private IUserService userService;
+
+    @Inject
+    private IEventService eventService;
+
+    private boolean isDatabaseConnected = false;
+
+    private final Map<String, VoiceConnection> clientConnections = new ConcurrentHashMap<>();
+
+    @Override
+    public void requestVoiceConnection(String identifier, Consumer<Optional<IVoiceConnection>> onRequestFinished) {
+        new Thread(
```

Comment:
Same as in ``VoiceConnection``. Please use executor services instead of instantiating new thread objects.